pennylane/operation.py

```diff
@@ -582,21 +582,79 @@ def decompose(self):
+        the sequence of diagonalizing gates implements the unitary :math:`U`.
+        In other words, the diagonalizing gates rotate the state into the eigenbasis
+        of this operator.
+
+        This is the static version of ``diagonalizing_gates``, which can be called
+        without creating an instance of the class.
+
+        .. note::
+
+            This method gets overwritten by subclasses to define the representation of a particular operator.
+            By default, this method should always take the operator's parameters, wires and hyperparameters as
+            inputs (even if the diagonalizing gates are independent of these values).
+
+            Alternatively, a custom signature can be defined, in which case the `diagonalizing_gates()`
+            method has to be overwritten to use the right signature.
+
+        Args:
+            params (list): trainable parameters of this operator, as stored in ``op.parameters``
+            wires (Iterable): trainable parameters of this operator, as stored in ``op.wires``
+            hyperparams (dict): non-trainable hyperparameters of this operator, as stored in ``op.hyperparameters``
+
+        Returns:
+            list[.Operator]: A list of operators.
+
+        *Example:*
+
+        >>> qml.PauliX.compute_diagonalizing_gates(wires="q1")
+        [Hadamard(wires=["q1"])]
+        """
+        return None
+
     # pylint:disable=no-self-use
     def diagonalizing_gates(self):
-        r"""Defines a partial representation of this operator as
-        an eigendecompisition.
+        r"""Defines a partial representation of this operator via
+        its eigendecomposition.
 
-        Multiplied together, the diagonalizing gates
-        form the unitary :math:`U` in `O = U \Sigma U^{\dagger}`, while
-        :math:`\Sigma` is a diagonal matrix containing the eigenvalues.
+        Given the eigendecomposition `O = U \Sigma U^{\dagger}` where
+        :math:`\Sigma` is a diagonal matrix containing the eigenvalues,
+        the sequence of diagonalizing gates implements the unitary :math:`U`.
+        In other words, the diagonalizing gates rotate the state into the eigenbasis
+        of this operator.
 
         Returns `None` if this operator does not define its diagonalizing gates.
 
+        .. note::
+
+            By default, this method calls the static method ``compute_diagonalizing_gates``,
+            which is used by subclasses to define the actual representation.
+            The call assumes that the static method has the signature ``(*params, wires, **hyperparams)``,
+            where ``params`` refers to ``op.parameters``, wires to ``op.wires`` and ``hyperparams``
+            to ``op.hyperparameters``, and ``op`` is an instance of this operator.
+
+            If a subclass overwrites ``compute_diagonalizing_gates`` to use a custom signature,
+            this method has to be likewise overwritten to respect that signature.
```

Comment:
Same here, is this enough to tell you how to overwrite?

Reply:
I am not too worried about a long note, as long as it explains well. Is the shorter one clearer or just shorter? :) 

I always appreciate when people give me detailed instructions in code...

Reply:
> Is the shorter one clearer or just shorter? :)

Perhaps I am conflating the two 😆 

My main motiviation (as mentioned in a previous comment) was:

- making this standout (the yellow background)
- explicitly calling out who this sentence is important for (those making their own operations). Users _using_ this operation can ignore it.